### PR TITLE
Allow configuring log format for Audit Logs

### PR DESCRIPTION
**Pull request by frankgh**
patch by Francisco Guerrero; reviewed by TBD for CASSANDRA-19792

### CODE REVIEW

**Review by smiklosovic — APPROVED** [review]

File: conf/cassandra.yaml [hunk 1]
```diff
@@ -1823,6 +1823,9 @@ unlogged_batch_across_partitions_warn_threshold: 10
 
 # Audit logging - Logs every incoming CQL command request, authentication to a node. See the docs
 # on audit_logging for full details about the various configuration options and production tips.
+# For BinAuditLogger and FileAuditLogger, the following optional parameters can be configured:
+# - key_value_separator: Specifies the key-value separator to use for the log message. (Defaults to ":")
```
Comment (tolbertam) [inline conf/cassandra.yaml:1827]: One thing that wasn't clear to me was where exactly these go, I tried:


```yaml
audit_logging_options:
    key_value_separator: "^"
    field_separator: "-"
```

and got :

```
ERROR [main] 2024-07-24 16:16:50,716 CassandraDaemon.java:910 - Exception encountered during startup: Invalid yaml. Please remove properties [field_separator, key_value_separator] from your cassandra.yaml
```

Instead can we change these comments to have them inline like:

```yaml
  logger:
    - class_name: BinAuditLogger
    # parameters:
    #  - key_value_separator: ...
```

Comment (tolbertam) [reply conf/cassandra.yaml:1827]: looks good now, thanks!

**Review by tolbertam — APPROVED** [review]
👍  🚀

**Comment by frankgh** [comment]
Closed via https://github.com/apache/cassandra/commit/25291ff3fd99f92cdb0a7d5d2125442282d42ff8

**Pull request closed without merging**